server/src/main/java/org/apache/iotdb/db/metadata/cache/TimeSeriesSchemaCache.java

```diff
@@ -159,6 +161,66 @@ public void computeValue(int index, SchemaCacheEntry value) {
     return indexOfMissingMeasurements;
   }
 
+  public Pair<List<Integer>, List<String>> computeSourceOfLogicalViewInCache(
+      ISchemaComputation schemaComputation) {
+    List<Integer> indexOfMissingMeasurements = new ArrayList<>();
+    List<String> missedPathStringList = new ArrayList<>();
+    final AtomicBoolean isFirstMeasurement = new AtomicBoolean(true);
+    Pair<Integer, Integer> beginToEnd = schemaComputation.getSizeOfLogicalViewSchemaListRecorded();
+    List<LogicalViewSchema> logicalViewSchemaList = schemaComputation.getLogicalViewSchemaList();
+    List<Integer> indexListOfLogicalViewPaths = schemaComputation.getIndexListOfLogicalViewPaths();
+    for (int i = beginToEnd.left; i < beginToEnd.right; i++) {
+      LogicalViewSchema logicalViewSchema = logicalViewSchemaList.get(i);
+      final int realIndex = indexListOfLogicalViewPaths.get(i);
+      final int recordMissingIndex = i;
+      if (!logicalViewSchema.isWritable()) {
+        throw new RuntimeException(
+            new UnsupportedOperationException(
+                "You can not insert into a logical view which is not alias series!"));
+      }
+      PartialPath fullPath = logicalViewSchema.getSourcePathIfWritable();
+      dualKeyCache.compute(
+          new IDualKeyCacheComputation<PartialPath, String, SchemaCacheEntry>() {
+            @Override
+            public PartialPath getFirstKey() {
+              return fullPath.getDevicePath();
+            }
+
+            @Override
+            public String[] getSecondKeyList() {
+              return new String[] {fullPath.getMeasurement()};
+            }
+
+            @Override
+            public void computeValue(int index, SchemaCacheEntry value) {
+              index = realIndex;
+              if (value == null) {
+                indexOfMissingMeasurements.add(recordMissingIndex);
+              } else {
+                if (isFirstMeasurement.get()) {
+                  schemaComputation.computeDevice(value.isAligned());
```

Comment:
We need record the alignment of source device and use it to split new insert statement. When inserting to a logical view with aligned source device, the insert statement shall be parsed as insertion with aligned values.

Reply:
fixed